app/src/src_nonfreenet/org/breezyweather/sources/cwa/CwaResultConverter.kt

```diff
@@ -218,39 +204,27 @@ private fun getNormals(
 // Concentrations of SO₂, NO₂, O₃ are given in ppb (and in ppm for CO).
 // We need to convert these figures to µg/m³ (and mg/m³ for CO).
 private fun getAirQuality(
-    result: CwaWeatherResult,
+    airQualityResult: CwaAirQualityResult?,
     temperature: Double?,
     pressure: Double?,
 ): AirQuality? {
-    return if (result.data?.aqi?.getOrNull(0) != null) {
+    return airQualityResult?.data?.aqi?.getOrNull(0)?.let {
         AirQuality(
-            pM25 = result.data.aqi[0].pm2_5?.toDoubleOrNull(),
-            pM10 = result.data.aqi[0].pm10?.toDoubleOrNull(),
-            sO2 = ppbToUgm3("SO2", result.data.aqi[0].so2?.toDoubleOrNull(), temperature, pressure),
-            nO2 = ppbToUgm3("NO2", result.data.aqi[0].no2?.toDoubleOrNull(), temperature, pressure),
-            o3 = ppbToUgm3("O3", result.data.aqi[0].o3?.toDoubleOrNull(), temperature, pressure),
-            cO = ppbToUgm3("CO", result.data.aqi[0].co?.toDoubleOrNull(), temperature, pressure)
+            pM25 = it.pm25?.toDoubleOrNull(),
+            pM10 = it.pm10?.toDoubleOrNull(),
```

Comment:
So PM is in Ug/m³ and others in PPB? How weird

Reply:
Yup. <https://airtw.moenv.gov.tw/ENG/default.aspx>
The CWA endpoints simply relay this information.